openupgrade_framework/odoo_patch/odoo/tests/loader.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 Hunki Enterprises BV (https://hunki-enterprises.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import importlib
+import inspect
+import os.path
+import sys
+
+from odoo.tests import loader
+from odoo.tools import config
+
+
+def _get_tests_modules(mod):
+    """
+    Make odoo load tests from openupgrade_scripts/scripts/$module/$version/tests
+    instead of the standard location
```

Comment:
It looks like https://github.com/odoo/odoo/blob/8c62f4a/odoo/tests/loader.py#L38-L57 makes it possible to load tests from upgrade paths, which is also the aim of this patch. Do you agree @hbrunn?

Reply:
Played around with this a bit, and I got it to load a test from the `--upgrade-path` (subpath `base/tests/test_migration_test.py`). Because it has a module path in `odoo.upgrade` instead of `odoo.addons` it does not get assigned the expected attributes in https://github.com/odoo/odoo/blob/aa84ccf/odoo/tests/common.py#L222-L227. The attributes therefore have to be assigned manually:

```
from odoo.tests import tagged, TransactionCase


@tagged("at_install", "openupgrade")
class TestMigrationTest(TransactionCase):
    def test_migration_test(self):
        import pdb
        pdb.set_trace()
        self.assertEqual(1, 0)

TestMigrationTest.test_module = "base"
TestMigrationTest.test_class = "TestMigrationTest"
TestMigrationTest.test_sequence = 0
```

After that, the test runs fine and can be selected with test tags `base:TestMigrationTest` or `openupgrade`.

Personally, I think that is acceptable given that tests are still very much underused and not for the faint of heart at this point.

Reply:
agreed the patch becomes obsolete in v17. We could provide a base class for running openupgrade tests, but that's for another PR